### DISCUSSION TITLE
Is there a way of dynamically loading and changing a texture (perferablly without the usage of resource packs?)

### DISCUSSION THREAD

**Discussion by NGSpace** [FabricMC/fabric-api#3708] [Mod Dev Support]
### Description

I am making a mod and I want the mod to be able to load (and also change) textures made by the user (perferablly through a png file) and draw it on the screen. but i was not able to find a way either by scrolling through the api or google...

#I found the answer!
```java
var sc = new SpriteContents(id, new SpriteDimensions(width, height), NativeImage.read(...), new ResourceMetadata.Builder().build());
Sprite sprite = new Sprite(id, sc, width, height, 0, 0){};
context.drawSprite(x, y, 0, width, height, sprite);
```

Note: The texture must be registered first!

**Comment by apple502j** [comment]
NativeImageBackedTexture.

**Reply by NGSpace** [reply]
is there a way to draw said texture on a DrawContext?
